kit/torch/data.py

```diff
@@ -68,13 +70,13 @@ def __init__(self, data_source: Sized, batch_size: int, shuffle: bool = True) ->
         self.data_source = data_source
         self.shuffle = shuffle
         self._dataset_size = len(data_source)
-        self.batch_size = min(batch_size, self._dataset_size)
+        self.batch_size = batch_size
 
     def _generate_idx_seq(self) -> Tensor:
         """Generate a random sequence of unique indexes."""
         if self.shuffle:
-            return torch.arange(self._dataset_size)
-        return torch.randperm(self._dataset_size)
+            return torch.randperm(self._dataset_size)
```

Comment:
shouldn't the object have a generator for random numbers? I get a bit confused as to what the best approach is for these

Reply:
You're right, this was lazy of me  - the class should have a generator attribute, following the example of torch's RandomSampler class.